src/base/bittorrent/session.cpp

```diff
@@ -2372,6 +2372,36 @@ void Session::setIPFilterFile(QString path)
     }
 }
 
+void Session::setBannedIPList(const QStringList &newList)
+{
+    if (newList == m_bannedIPs)
+        return; // do nothing
+    // malformed IP strings will be silently filtered out
+    QStringList filteredList;
+    foreach (const QString &ip, newList) {
+        boost::system::error_code ec;
+        libt::address::from_string(ip.toLatin1().constData(), ec); // only error code needed
+        if (!ec) {
+            filteredList << ip;
+        }
+        else {
+            Logger::instance()->addMessage(
+                tr("During saving list of banned IP addresses \"%1\" did not pass validation and was not saved.")
```

Comment:
In "saving list" pair the gerund does not pose an action. An article is needed between them, i.e. "saving the list". Consider
"'%1' is not a valid IP address and was rejected while applying the list of banned addresses."